web/modules/custom/usagov_benefit_category_search/js/benefitSearch.js

```diff
@@ -444,14 +446,19 @@ ${benefit.term_node_tid}
 
     let elt = document.createElement('template');
     elt.innerHTML = `<div class="usa-alert usa-alert--slim usa-alert--error margin-bottom-4" aria-live=assertive>
-        <div class="usa_alert__body">
+        <div class="usa_alert__body" data-analytics="errorMessage">
            <h4 class="usa-alert__heading padding-left-6">${myself.labels.emptyCategoryError}</h4>
         </div>
     </div>`;
     myself.form.querySelector('.alert-container').prepend(elt.content);
 
     const fieldset = myself.form.querySelector('div[role="group"]');
     fieldset.classList.add('benefits-category-error');
+
+    // sending data to GTM when the error message appears
+    if (document.querySelector('[data-analytics="errorMessage"]')) {
+      dataLayer.push({'event': 'category_apply_error'});
+    }
```

Comment:
Same, we should check if dataLayer exisits
```suggestion
    if (dataLayer != null && document.querySelector('[data-analytics="errorMessage"]')) {
      dataLayer.push({'event': 'category_apply_error'});
    }
```

Reply:
good call.. i updated with an if statement